### PR TITLE
Only report None tokens as EOF

### PR DESCRIPTION
**Pull request by mandolaerik**
SIMICS-19465

### CODE REVIEW

File: py/dml/dmlparse.py [hunk 1]
```diff
@@ -2512,15 +2512,15 @@ def reserved(t):
...
-        raise ESYNTAX(DumpableSite(t.lexer.file_info, t.lexpos), value, None)
-    else:
+    if t is None:
         # raise ESYNTAX(...) would have made sense here, but there
         # is no clean way to retrieve the file (because t is
         # None). Therefore, push the error reporting responsibility to
         # whoever invokes the parser.
         raise UnexpectedEOF()
+    else:
+        value = str(t.value)
```
Comment (lwaern-intel) [inline py/dml/dmlparse.py:2522]: What value did the previous `t.value` check have? You no longer perform it, so we must consider if it contributed anything.

Comment (mandolaerik) [reply py/dml/dmlparse.py:2522]: Likely no value. We should not preserve things out of superstition. Worst case is if t.value would be a false value that doesn't display nicely, and in this case we will move from an anonymous EOF error to an error that points at the correct token but presents that token as garbage. That's an improvement.

Comment (lwaern-intel) [reply py/dml/dmlparse.py:2522]: I'm inclined to agree.

**Comment by syssimics** [comment]
Verification [#10414297](https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=10414297): pass

**Review by JonatanWaern — APPROVED** [review]

**Pull request merged**